DNN Platform/Library/Entities/Profile/ProfileController.cs

```diff
@@ -313,6 +314,31 @@ public static void DeletePropertyDefinition(ProfilePropertyDefinition definition
             _dataProvider.DeletePropertyDefinition(definition.PropertyDefinitionId);
             EventLogController.Instance.AddLog(definition, PortalController.Instance.GetCurrentPortalSettings(), UserController.Instance.GetCurrentUserInfo().UserID, "", EventLogController.EventLogType.PROFILEPROPERTY_DELETED);
             ClearProfileDefinitionCache(definition.PortalId);
+            ClearAllUsersInfoProfileCacheByPortal(definition.PortalId);
+
+        }
+
+        /// -----------------------------------------------------------------------------
+        /// <summary>
+        /// Clear profiles of all users by portal Id
+        /// </summary>
+        /// -----------------------------------------------------------------------------
+        public static void ClearAllUsersInfoProfileCacheByPortal(int portalId)
+        {
+            DataCache.ClearCache(GetUserInfoPrefix(portalId));
+            DataCache.ClearCache(GetUserProfilePrefix(portalId));
+        }
+
+        private static string GetUserInfoPrefix(int portalId)
+        {
+            var prefix = DataCache.UserCacheKey.Remove(DataCache.UserCacheKey.LastIndexOf('|') + 1);
+            return string.Format(prefix, portalId);
```

Comment:
let's get rid of hard coded separator '|' and simplify this to be string.Format(DataCache.UserCacheKey, portalId, "")